imap_processing/spice/pointing_frame_handler.py

```diff
@@ -0,0 +1,242 @@
+"""
+Generate Pointing Frame.
+
+Notes
+-----
+Kernels that are required to run this code:
+1. imap_science_0001.tf - pointing frame kernel
+2. imap_sclk_0000.tsc - spacecraft clock kernel
+3. imap_wkcp.tf - spacecraft frame kernel
+4. naif0012.tls - standard NAIF leapsecond kernel
+5. imap_spin.bc - test attitude kernel generated by simulate_imap_attitude.m.
+These need to be placed in tests/pointing_frame/test_data.
+
+References
+----------
+https://spiceypy.readthedocs.io/en/main/documentation.html.
+"""
+
+import logging
+import os
+from pathlib import Path
+
+import numpy as np
+import spiceypy as spice
+from numpy.typing import NDArray
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# TODO : Add multiple pointings to the pointing frame.
+
+
+def get_et_times(ck_kernel: str) -> tuple[float, float, np.ndarray]:
+    """
+    Create the pointing frame.
+
+    Parameters
+    ----------
+    ck_kernel : str
+        Path of ck_kernel used to create the pointing frame.
+
+    Returns
+    -------
+    et_start : float
+        Start time of ck_kernel.
+    et_end : float
+        End time of ck_kernel.
+    et_times : numpy.ndarray
+        Array of times between et_start and et_end.
+    """
+    # Get the spacecraft ID.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.gipool
+    id_imap_spacecraft = spice.gipool("FRAME_IMAP_SPACECRAFT", 0, 1)
+
+    # TODO: Queried pointing start and stop times here.
+
+    # Get the coverage window
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.ckcov
+    cover = spice.ckcov(ck_kernel, int(id_imap_spacecraft), True, "SEGMENT", 0, "TDB")
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.wnfetd
+    et_start, et_end = spice.wnfetd(cover, 0)
+    # Assumes a pointing is ~ 1 day.
+    # 1 spin/15 seconds * 86400 seconds/pointing * 10 quaternions / spin
+    et_times = np.linspace(et_start, et_end, 57600)
```

Comment:
Can you make this dynamic instead of assuming 24 hour pointing...
```suggestion
    # Sample the attitude ~10 times per 15-second spin
    num_samples = (et_end - et_start) / 15 * 10
    et_times = np.linspace(et_start, et_end, )
```